### ISSUE TITLE
长文本的支持，有没有实现方法？

### ISSUE THREAD

**Issue by EricKong1985** [Plachtaa/VALL-E-X#7]
本人对于深度学习不是很懂，出笑话不要介意
我看了bark对于长文本的实现，
https://github.com/suno-ai/bark/blob/main/notebooks/long_form_generation.ipynb
然后我改造了一下ui的代码，
import nltk
nltk.download('punkt')

@torch.no_grad()
def infer_from_prompt(text, language, accent, prompt_file):
    clear_prompts()
    model.to(device)
    # text to synthesize
    lang_token = langdropdown2token[language]
    lang = token2lang[lang_token]
    sentences = nltk.sent_tokenize(text)
    silence = np.zeros(int(0.25 * SAMPLE_RATE))  # quarter second of silence
    pieces = []
    for sentence in sentences:
        print("\n生成语音中:" + sentence + "\n")
    audio_array,t = get_voice(accent, lang, lang_token, prompt_file, sentence)
    pieces += [audio_array, silence.copy()]
    normalized_pieces = np.concatenate(pieces)
    normalized_pieces = (normalized_pieces * 32767 / np.max(np.abs(normalized_pieces))).astype(np.int16)
    # save audio to disk
    model.to('cuda')
    torch.cuda.empty_cache()
    message = f"sythesized text: {text}"
    return message, (24000, normalized_pieces)

def get_voice(accent, lang, lang_token, prompt_file, text):
    text = lang_token + text + lang_token
    print("lang:" + text)
    # load prompt
    prompt_data = np.load(prompt_file.name)
    audio_prompts = prompt_data['audio_tokens']
    text_prompts = prompt_data['text_tokens']
    lang_pr = prompt_data['lang_code']
    lang_pr = code2lang[int(lang_pr)]
    # numpy to tensor
    audio_prompts = torch.tensor(audio_prompts).type(torch.int32).to(device)
    text_prompts = torch.tensor(text_prompts).type(torch.int32)
    enroll_x_lens = text_prompts.shape[-1]
    logging.info(f"synthesize text: {text}")
    phone_tokens, langs = text_tokenizer.tokenize(text=f"_{text}".strip())
    print("langs：" + str(langs))
    text_tokens, text_tokens_lens = text_collater(
        [
            phone_tokens
        ]
    )
    text_tokens = torch.cat([text_prompts, text_tokens], dim=-1)
    text_tokens_lens += enroll_x_lens
    # accent control
    lang = lang if accent == "no-accent" else token2lang[langdropdown2token[accent]]
    else_lang = langs if accent == "no-accent" else lang
    print("else_lang:" + else_lang)
    encoded_frames = model.inference(
        text_tokens.to(device),
        text_tokens_lens.to(device),
        audio_prompts,
        enroll_x_lens=enroll_x_lens,
        top_k=-100,
        temperature=1,
        prompt_language=lang_pr,
        text_language=else_lang,
    )
    samples = audio_tokenizer.decode(
        [(encoded_frames.transpose(2, 1), None)]
    )
    audio_data = samples[0][0].cpu().numpy()
    return audio_data, text
但是好像好像分拆句子不成功，看有没有大神，路过一起解决这个问题，争取这个项目早日支持长文本呢？

**Comment by Plachtaa** [comment]
一般来说，不同语言需要不同的库来实现切句子脚本，bark提供的nltk库只能切英文。
总体而言，切句子脚本并不难实现，但是可能要过一段时间才会更新

**Comment by EricKong1985** [comment]
有计划吗？ 好像这几天都没有更新

**Comment by Plachtaa** [comment]
别急
